ocaml-backend/src/main/java/org/kframework/backend/ocaml/DefinitionToOcaml.java

```diff
@@ -477,11 +480,34 @@ private void matchPatternAndPrint(StringBuilder sb) {
         sb.append("let _ = (try print_subst_binary subst (try_match res res (-1)) with Stuck c -> output_string subst \"\\x00\\x00\\x00\\x00\")\n");
     }
 
+    private void generateMain(StringBuilder sb) {
+        indentDepth = 0;
+        indented(sb).append("let main =\n");
```

Comment:
this seems very weird and unnecessary to me. Why was this done? note that you haven't actually declared a function here.

Reply:
I did that because I could not find any way to use exceptions at toplevel. So I just declare a function and then call it.